pybop/optimisers/pints_optimisers.py

```diff
@@ -116,12 +117,16 @@ class PSO(pints.PSO):
     """
 
     def __init__(self, x0, sigma0=0.1, bounds=None):
-        if bounds is not None:
+        if bounds is None:
+            self.boundaries = None
+        elif not np.all(np.isfinite(sum(bounds.values(), []))):
```

Comment:
I think this might do what you are wanting in a more pythonic way.
```suggestion
        elif not all(np.isfinite(value) for sublist in bounds.values() for value in sublist):
```